packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -105,13 +158,12 @@ async function findAllHotswappableChanges(
       continue;
     }
 
-    // no hotswappable changes found, so any REQUIRES_FULL_DEPLOYMENTs imply a non-hotswappable change
-    for (const result of hotswapDetectionResults) {
-      if (result === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT) {
-        foundNonHotswappableChange = true;
-      }
+    // no hotswappable changes found, so at least one IRRELEVANT means we can ignore this change;
+    // otherwise, all answers are REQUIRES_FULL_DEPLOYMENT, so this means we can't hotswap this change,
+    // and have to do a full deployment instead
+    if (hotswapDetectionResults.every(hdr => hdr === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT)) {
```

Comment:
This condition might technically be correct, but it reads weird to me:

* Code says: "we only need to need to deploy if *everything* says we need to deploy"
* My expectation: "we need to deploy as soon as *something* says we need to deploy"

The code doesn't jive with my expectations, making me do a double- and triple-take here.

Isn't the following more in line with the intent of what you're trying to convey:

```ts
if (!hotswapDetectionResults.some(hdr => hdr === ChangeHotswapImpact.IRRELEVANT)) {
```

"we only need to deploy as long as *nothing* says the change is irrelevant"

I get that those statements are equivalent for the machine(*), but they're not equivalent to a human reading them.


> (*) given the current possible enum values, but who knows what we might add in the future? We might want to be robust against future extensions

Reply:
By the way, an even more accurate modeling I think would be the following:

```ts
if (results.filter(hdr => hdr !== IRRELEVANT).some(hdr => hdr === REQUIRES_FULL_DEPLOYMENT)) {
  doFullDeployment();
}
```

"After taking out all the IRRELEVANTs, if something is still asking for a full deployment then let's do that"

Reply:
Or, have every function not return a `ChangeHotswapResult` but a `ChangeHotswapResult[]`, and return an empty array instead of `IRRELEVANT`.

Reply:
Changed the condition to `!hotswapDetectionResults.some(hdr => hdr === ChangeHotswapImpact.IRRELEVANT)` - I actually debated between that, and the `every()` condition that is there in this version of the PR, so I like the change.

`hotswapDetectionResults.filter(hdr => hdr !== IRRELEVANT).some(hdr => hdr === REQUIRES_FULL_DEPLOYMENT)` actually means something different - "if there's at least one REQUIRES_FULL_DEPLOYMENT, we can't hotswap", which is the current logic, which this PR wants to change. So, it would have to be `!hotswapDetectionResults.filter(hdr => hdr !== REQUIRES_FULL_DEPLOYMENT).some(hdr => hdr === IRRELEVANT)`, which I don't think reads that well.